packages/earn-controller/src/EarnController.ts

```diff
@@ -0,0 +1,334 @@
+  AccountsControllerSelectedAccountChangeEvent,
+} from '@metamask/accounts-controller';
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedControllerMessenger,
+  StateMetadata,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import { convertHexToDecimal } from '@metamask/controller-utils';
+import type { NetworkControllerNetworkDidChangeEvent } from '@metamask/network-controller';
+import type {
+  NetworkControllerGetNetworkClientByIdAction,
+  NetworkControllerGetStateAction,
+} from '@metamask/network-controller';
+import {
+  StakeSdk,
+  StakingApiService,
+  type PooledStake,
+  type StakeSdkConfig,
+  type VaultData,
+} from '@metamask/stake-sdk';
+
+export const controllerName = 'EarnController';
+
+// === Enums and Identifiers ===
+export enum EarnProductType {
+  POOLED_STAKING = 'pooled_staking',
+  STABLECOIN_LENDING = 'stablecoin_lending',
+}
+
+// === Product Types ===
+export type PooledStakingProduct = {
+  pooledStakes: PooledStake;
+  exchangeRate: string;
+  vaultData: VaultData;
+  isEligible: boolean;
+};
+
+export type StablecoinLendingProduct = {
+  vaults: StablecoinVault[];
+};
+
+export type StablecoinVault = {
+  symbol: string;
+  name: string;
+  chainId: number;
+  tokenAddress: string;
+  vaultAddress: string;
+  currentAPY: string;
+  supply: string;
+  liquidity: string;
+};
+
+/**
+ * Metadata for the EarnController.
+ */
+export const earnControllerMetadata: StateMetadata<EarnControllerState> = {
+  [EarnProductType.POOLED_STAKING]: {
+    persist: true,
+    anonymous: false,
+  },
+  [EarnProductType.STABLECOIN_LENDING]: {
+    persist: true,
+    anonymous: false,
+  },
+  lastUpdated: {
+    persist: false,
+    anonymous: true,
+  },
+};
+
+// === State Types ===
+export type EarnControllerState = {
+  [EarnProductType.POOLED_STAKING]: PooledStakingProduct;
+  [EarnProductType.STABLECOIN_LENDING]?: StablecoinLendingProduct;
+  lastUpdated: number;
+};
+
+// === Default State ===
+const DEFAULT_STABLECOIN_VAULT: StablecoinVault = {
+  symbol: '',
+  name: '',
+  chainId: 0,
+  tokenAddress: '',
+  vaultAddress: '',
+  currentAPY: '0',
+  supply: '0',
+  liquidity: '0',
+};
+
+/**
+ * Gets the default state for the EarnController.
+ *
+ * @returns The default EarnController state.
+ */
+export function getDefaultEarnControllerState(): EarnControllerState {
+  return {
+    [EarnProductType.POOLED_STAKING]: {
+      pooledStakes: {
+        account: '',
+        lifetimeRewards: '0',
+        assets: '0',
+        exitRequests: [],
+      },
+      exchangeRate: '1',
+      vaultData: {
+        apy: '0',
+        capacity: '0',
+        feePercent: 0,
+        totalAssets: '0',
+        vaultAddress: '0x0000000000000000000000000000000000000000',
+      },
+      isEligible: false,
+    },
+    [EarnProductType.STABLECOIN_LENDING]: {
+      vaults: [DEFAULT_STABLECOIN_VAULT],
```

Comment:
Should we add an isEligible key for stablecoin lending? We discussed that right now we would hide both if the user is not eligible for pooled staking but it could change in the future.

Alternatively, if we are going to stick with one key for the earn products generally, is there a more global place for it in the state structure, possibly at the level of `lastUpdated`?